bin/extract_api_surface.py

```
@@ -0,0 +1,207 @@
+    methods = {}
+    for node in ast.walk(tree):
+        if isinstance(node, ast.ClassDef) and node.name == class_name:
+            for item in node.body:
+                if isinstance(item, (ast.FunctionDef, ast.AsyncFunctionDef)):
+                    if item.name.startswith("_"):
+                        continue
+                    methods[item.name] = _signature_from_function(item)
+    return methods
+
+
+def _find_source_file(pkg_dir, module_name):
```

Ruff (0.15.7)

[warning] Missing return type annotation for private function `_find_source_file`

(ANN202)